nav2_map_server/README.md

```diff
@@ -62,83 +136,336 @@ occupied_thresh: 0.65
+# Returned map is only valid if result equals RESULT_SUCCESS
+nav_msgs/OccupancyGrid map
+uint8 result
 ```
+
+2. 2D Map Saver
+- SaveMap
+	- nav2_msgs/srv/SaveMap.srv
+```yaml
+# URL of map resource
+# Can be an absolute path to a file: file:///path/to/maps/floor1.yaml
+# Or, relative to a ROS package: package://my_ros_package/maps/floor2.yaml
+string map_topic
+string map_url
+# Constants for image_format. Supported formats: pgm, png, bmp
+string image_format
+# Map modes: trinary, scale or raw
+string map_mode
+# Thresholds. Values in range of [0.0 .. 1.0]
+float32 free_thresh
+float32 occupied_thresh
+---
+bool result
+```
+
+### Point Cloud services
+1. 3D Map server
+- GetMap
+	- nav2_msgs/srv/GetMap3D.srv
+```yaml
+# Get the map as a sensor_msgs/PointCloud2
+---
+# The current map hosted by this map service.
+sensor_msgs/PointCloud2 map
+geometry_msgs/Pose origin
+```
+
+- LoadMap
+	- nav2_msgs/srv/LoadMap3D.srv
+```yaml
+# URL of map resource
+# Can be an absolute path to a file: file:///path/to/maps/floor1.yaml
+# Or, relative to a ROS package: package://my_ros_package/maps/floor2.yaml
+string map_url
+---
+# Result code definitions
+uint8 RESULT_SUCCESS=0
+uint8 RESULT_MAP_DOES_NOT_EXIST=1
+uint8 RESULT_INVALID_MAP_DATA=2
+uint8 RESULT_INVALID_MAP_METADATA=3
+uint8 RESULT_UNDEFINED_FAILURE=255
+# Returned map is only valid if result equals RESULT_SUCCESS
+sensor_msgs/PointCloud2 map
+geometry_msgs/Pose origin
+uint8 result
+```
+2. 3D Map saver
+- SaveMap
+	- nav2_msgs/srv/SaveMap3D.srv
+```yaml
+# URL of map resource
+# Can be an absolute path to a file: file:///path/to/maps/floor1.yaml
+# Or, relative to a ROS package: package://my_ros_package/maps/floor2.yaml
+string map_topic
+string origin_topic
+string map_url
+# as_binary: If true then map will be saved as a binary file
+# else ASCII format will be used.
+bool as_binary
+# Constants for image_format. Supported formats: pcd, ply
+string file_format
+---
+bool result
+```
+
+---
+### Service Usage
+ 
+For using these services `map_server/map_saver` should be launched as a continuously running `nav2::LifecycleNode` node. In addition to the CLI, Map Saver has a functionality of server handling incoming services. To run Map Saver in a server mode, provided launch-files could be used.
+ 
+| LaunchFile | Assiciated executable | Functionality | Type | Map Type
+| :-------------: | :-----------------: | :--------------: | :----: | :---: |
+| map_saver_server_2d.launch.py | main_server_2d.cpp | Map Saver | LifcyleNode | OccupancyGrid
+| map_saver_server_3d.launch.py | main_server_3d.cpp |  Map Saver | LifcyleNode | PointCloud2
+
+```shell
+$ ros2 launch nav2_map_server map_saver_server_2d.launch.py
+```
+```shell
+$ ros2 launch nav2_map_server map_saver_server_3d.launch.py
+```
+
+ Once the services are online, any program can call the running services as shown in the examples below: (use `ros2 node list` to get the list of nodes and `ros2 service list` to get the running services.)
```

Comment:
`ros2 node list` and `ros2 service list` are well-known commands. Please remove the clarification in parentheses.